ui/src/components/layout/FlowCanvas.tsx

```diff
@@ -173,24 +165,30 @@ const FlowCanvas: React.FC<Props> = () => {
     }
   }
 
-  function handleAddNode(item: PathedData) {
+  async function handleAddNode(item: PathedData) {
     // Item is from the form entered in the modal
     // and has already been stored on the DHT by this point
 
+    // TODO: if we had an interface with a `name` property, we wouldn't need this.
+    const T = ObjectTypeMap[type];
+    const transformer = ObjectTransformations[type];
+    const data: typeof T = transformer(item);
+
     // Create react flow node object
     const id = Guid.raw();
     const node = {
       id: id,
-      vfPath: item.path,
+      vfPath: data.path,
       type,
       position: currentPosition,
       // TODO: The is wrong for process, since it ends up saying "ProcessSpecification"
-      data: { id: id, label: `${type.charAt(0).toUpperCase()}${type.slice(1)}`, name: currentNodeName }
+      data: { id: id, label: `${type.charAt(0).toUpperCase()}${type.slice(1)}`, name: data.name }
```

Comment:
This whole way of doing this feels wrong. I'd like to somehow add the ability to get a cursor and have the `DisplayNode` object itself have a getter for the `data` property and it could compute it automatically from the path. That way, it could also use the automatically generated `id` instead of manually setting it. However, i'd need to get the `DataStore` in a non-async manner.